cmd/minikube/cmd/stop.go

```diff
@@ -36,7 +36,12 @@ itself, leaving all files intact. The cluster can be started again with the "sta
 		fmt.Println("Stopping local OpenShift cluster...")
 		api := libmachine.NewClient(constants.Minipath, constants.MakeMiniPath("certs"))
 		defer api.Close()
-
+		if constants.Registrator.CompatibleWithDistribution(){
```

Comment:
The problem with using configuration is that atm I am not 100% yet how this works. This probably needs some side investigations first. Happy to do this, if you want to go down this path. It would also help us in the case where you call _delete_ on a stopped VM. In this case, we would like to know whether this VM is registered. We could get this from the config. If it is, we can print out a warning that you first need to start the VM, before deleting it so that un-registration can happen. From this perspective config setting makes most sense.

To move this along though, I think it is easiest to run the detector at this point in time again.